python/flink_agents/runtime/remote_execution_environment.py

```diff
@@ -0,0 +1,165 @@
+        """
+        if isinstance(input, KeyedStream):
+            self.__input = input.process(
+                MapKeyedProcessFunctionAdapter(),
+                output_type=RowTypeInfo(
+                    [PickledBytesTypeInfo(), PickledBytesTypeInfo()]
+                ),
+            )
+        else:
+            self.__input = input.map(
+                lambda x: Row(
+                    key_selector.get_key(x)
+                    if key_selector is not None
+                    else uuid.uuid4(),
+                    InputEvent(input=x),
+                ),
+                output_type=RowTypeInfo(
+                    [PickledBytesTypeInfo(), PickledBytesTypeInfo()]
+                ),
+            )
+        return self
+
+    def apply(self, workflow: Workflow) -> "AgentsExecutionEnvironment":
+        """Set workflow of execution environment.
+
+        Parameters
+        ----------
+        workflow : Workflow
+            The workflow user defined to run in execution environment.
+        """
+        self.__workflow_plan = WorkflowPlan.from_workflow(workflow)
+        return self
+
+    def to_datastream(self) -> DataStream:
+        """Get outputs of workflow execution. Used for remote execution.
+
+        Returns:
+        -------
+        DataStream
+            Outputs of workflow execution.
+        """
+        j_data_stream_output = invoke_method(
+            None,
+            "org.apache.flink.agents.runtime.FlinkAgent",
+            "connectToWorkflow",
+            [
+                self.__input._j_data_stream,
+                self.__workflow_plan.model_dump_json(serialize_as_any=True),
+            ],
+            [
+                "org.apache.flink.streaming.api.datastream.DataStream",
+                "java.lang.String",
+            ],
+        )
+        output_stream = DataStream(j_data_stream_output)
+        self.__output = output_stream.map(lambda x: cloudpickle.loads(x).output)
+        return self.__output
+
+    def execute(self) -> None:
+        """Execute agents.
+
+        This method is not supported for remote execution environments.
+        """
+        msg = "RemoteExecutionEnvironment does not support execute locally."
+        raise NotImplementedError(msg)
+
+    def from_list(self, input: List[Dict[str, Any]]) -> "AgentsExecutionEnvironment":
+        """Set input list of workflow execution.
+
+        This method is not supported for remote execution environments.
+        """
+        msg = "RemoteExecutionEnvironment does not support from_list."
+        raise NotImplementedError(msg)
+
+    def to_list(self) -> List[Dict[str, Any]]:
+        """Get output list of workflow execution.
+
+        This method is not supported for remote execution environments.
+        """
+        msg = "RemoteExecutionEnvironment does not support to_list."
+        raise NotImplementedError(msg)
+
+
+def get_execution_environment(**kwargs: Dict[str, Any]) -> AgentsExecutionEnvironment:
```

Comment:
Might be better to use a name different from it's supper class. Maybe `create_instance()`?